src/axom/quest/io/STLWriter.hpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef QUEST_STLWRITER_HPP_
+#define QUEST_STLWRITER_HPP_
+
+// Axom includes
+#include "axom/config.hpp"
+#include "axom/core/Macros.hpp"
+#include "axom/mint/mesh/Mesh.hpp"
+
+#include <string>
+
+namespace axom
+{
+namespace quest
+{
+/*!
+ * \class STLWriter
+ *
+ * \brief A simple STL writer for the faces of Mint meshes.
```

Comment:
Future leaning suggestion: I think we'll want to update the readers/writers to work with blueprint/BUMP. 

Should we emphasize that this works on collections of triangles (or perhaps planar polygons) rather than the input representation (mint)?

```suggestion
 * \brief A simple STL writer for a collection of triangular faces with limited support for planar polygons
```

Obviously, we can change this in the future, so feel free to ignore this suggestion in this PR.

Reply:
I agree that the readers/writers can eventually provide Conduit interfaces and use Bump internally for accessing those meshes. Future PR.

```
// Future
struct STLWriter {
//...
#if defined(AXOM_USE_CONDUIT)
   void write(const conduit::Node &mesh);
#endif
//...
};
```